### ISSUE TITLE
Unable to Log into Disqus

### ISSUE THREAD

**Issue by Sangeppato** [bromite/bromite#470]
### Bromite version

Version: 79.0.3945.123
Arch: arm64
Android version: 10
Device model: Xiaomi Mi A2

### Is this bug about the SystemWebView?
No

### Is the bug reproducible with latest version?
Yes

### Can the bug be reproduced with the latest Chomium version?
No

### Is the bug a crash?
No

### Describe the bug
When trying to Log into Disqus in an article's comment section, an error is generated: "There was an error submitting the form.", resulting in no login.

### Steps to reproduce the bug

1. Go to any article's comment section that uses Disqus
2. Try to login from there
3. See error

### Expected behavior
You should be able to login

### Screenshots
![Screenshot_20200119-150917](https://user-images.githubusercontent.com/37263823/72682488-bd7cc980-3acd-11ea-9a6b-d74a172bd8da.png)

Note that it's possible to log into Disqus from the Disqus website

**Comment by csagan5** [comment]
This might be due to trackers being blocked. Have you tested disabling the adblocker?

**Comment by randomshell** [comment]
@csagan5 it's the same issue as #464. In fact the title of the tab is 'CSRF verification failed (403)' because disqus uses referes too but it should use other methods for csrf protection.

And this is the second issue opened in a week for the same problem on different websites :thinking:

**Comment by csagan5** [comment]
@randomshell sorry, I missed out the comments there. 

There are no patches which directly changes how referers are used (and none possibly related which were introduced recently), I am however thinking that this patch might be responsible: https://github.com/bromite/bromite/blob/79.0.3945.123/build/patches/Enable-site-per-process-isolation-for-devices-with-enough-memory.patch

@randomshell how did you specifically figure out that it's a referrers problem?

**Comment by csagan5** [comment]
I think I found the bug; the flag introduced in [this patch](https://github.com/bromite/bromite/blob/master/build/patches/Reduce-HTTP-headers-in-DoH-requests-to-bare-minimum.patch)
 is activated also for `SUPPORT_ASYNC_REVALIDATION`, while it should not.

The bug has been introduced with 79.0.3945.94; I will provide a bugfix with next release.

@randomshell thanks again for digging out the correlation with referers

**Comment by randomshell** [comment]
> There are no patches which directly change how referers are used

This explains why I couldn't find why bromite doesn't send them.

> this patch might be responsible

I'm not sure about site per process. Is it hardcoded? I mean, I tried disabling the flag but disqus doesn't work anyway.
Steps I did:
1)`chrome://process-internals/#general` shows `Site Isolation mode: Site Per Process, Isolate Password Sites`.
2)Set `chrome://flags/#site-isolation-trial-opt-out` to `Disabled (not recommended) `. Now `chrome://process-internals/#general` shows `Site Isolation mode: Disabled`.
3)disqus shows same error

 
> how did you specifically figure out that it's a referrers problem?

I used chromium developer tools with adb debugging and looked at the difference between desktop http requests and bromite and saw the Referer header was missing. Checking the referer is common on login pages for csrf checking. I disabled it on desktop with an extension and confirmed this was the problem.
